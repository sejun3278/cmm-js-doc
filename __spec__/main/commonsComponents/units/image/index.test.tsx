import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import _Image from "src/main/commonsComponents/units/image";
import { useRouter } from "next/router";
import { MutableRefObject, useRef } from "react";

// 가짜 router 만들기
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));
const push = jest.fn();
(useRouter as jest.Mock).mockImplementation(() => ({
  push,
}));

// 가짜 ref 만들기
// jest.mock("react", () => ({
//   useRef: jest.fn(),
// }));
// (useRef as jest.Mock).mockImplementation(() => {});

export default it("Image Units Page Sanpshot", () => {
  const _ref = useRef() as MutableRefObject<HTMLImageElement>;
  const component = render(
    <_Image src="/images/" className="jest_image" _ref={_ref} />
  );
  // @ts-ignore
  expect(component.container).toMatchSnapshot();
});
