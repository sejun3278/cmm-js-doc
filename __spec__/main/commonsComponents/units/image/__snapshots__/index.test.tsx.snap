// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Units Page Sanpshot 1`] = `
<div>
  <img
    class="_image_ jest_image css-msby2z"
    src="/images/"
  />
</div>
`;
